DRODLib/OrbUtil.cpp

```diff
@@ -0,0 +1,130 @@
+// $Id$
+
+/* ***** BEGIN LICENSE BLOCK *****
+* Version: MPL 1.1
+*
+* The contents of this file are subject to the Mozilla Public License Version
+* 1.1 (the "License"); you may not use this file except in compliance with
+* the License. You may obtain a copy of the License at
+* http://www.mozilla.org/MPL/
+*
+* Software distributed under the License is distributed on an "AS IS" basis,
+* WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+* for the specific language governing rights and limitations under the
+* License.
+*
+* The Original Code is Deadly Rooms of Death.
+*
+* The Initial Developer of the Original Code is
+* Caravel Software.
+* Portions created by the Initial Developer are Copyright (C) 1995, 1996,
+* 1997, 2000, 2001, 2002, 2005, 2016 Caravel Software. All Rights Reserved.
+*
+* Contributor(s):
+*
+* ***** END LICENSE BLOCK ***** */
+
+#include "OrbUtil.h"
+
+void GetOrbConnections(CDbRoom& room, CCoordSet doorCoords, vector<COrbData*>& outOrbData) {
+
+	for (UINT orb = 0; orb < room.orbs.size(); ++orb)
+	{
+		COrbData* pData = room.orbs[orb];
+		for (UINT agent = 0; agent < pData->agents.size(); ++agent)
+		{
+			COrbAgentData* pAgentData = pData->agents[agent];
+			if (doorCoords.has(pAgentData->wX, pAgentData->wY))
+			{
+				outOrbData.push_back(pData);
+			}
+		}
+	}
+}
+
+void MergeOrbConnections(CDbRoom& room, CCoordSet doorCoords, COrbData* pOrbData) {
```

Comment:
Add comment describing the algorithm in human-understandable form, e.g., something like what you wrote up in the forum topic.